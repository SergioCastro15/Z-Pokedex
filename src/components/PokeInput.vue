<template>
  <div>
    <b-form-input
      v-model="text"
      placeholder="Search Pokemon"
      class="mt-5"
      ></b-form-input>
    <b-button
      variant="success"
      class="mt-2"
      @click="onSearchPokemon"
    >Search
    </b-button>
  </div>
</template>

<script>
export default {
  name: 'PokeInput',
  data() {
    return {
      text: '',
    };
  },
  methods: {
    onSearchPokemon() {
      this.$emit('triggerSearchPokemon', this.text);
    },
  },
};
</script>
