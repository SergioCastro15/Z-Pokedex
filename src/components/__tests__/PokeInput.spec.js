import { shallowMount } from '@vue/test-utils';
import PokeInput from '../PokeInput.vue';

describe('PokeInput component', () => {
  it('render', () => {
    const wrapper = shallowMount(PokeInput);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should send a emit to father element', () => {
    const localThis = {
      text: 'pikachu',
      $emit(method, text) {
        expect(method).toBe('triggerSearchPokemon');
        expect(typeof text).toBe('string');
      },
    };
    PokeInput.methods.onSearchPokemon.call(localThis);
  });
});
