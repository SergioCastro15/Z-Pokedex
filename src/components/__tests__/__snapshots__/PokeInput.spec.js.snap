// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokeInput component render 1`] = `
<div>
  <b-form-input
    class="mt-5"
    placeholder="Search Pokemon"
  />
   
  <b-button
    class="mt-2"
    variant="success"
  >
    Search
  
  </b-button>
</div>
`;
